lib/std/io/reader.zig

```diff
@@ -231,10 +231,16 @@ pub fn Reader(
             return mem.readVarInt(ReturnType, bytes, endian);
         }
 
-        pub fn skipBytes(self: Self, num_bytes: u64) !void {
-            var i: u64 = 0;
-            while (i < num_bytes) : (i += 1) {
-                _ = try self.readByte();
+        /// Reads `num_bytes` bytes from the stream and discards them
+        pub fn skipBytes(self: Self, num_bytes: usize) !void {
+            const buf_size = 512;
```

Comment:
Is 512 heuristically chosen, or is there some further thought put in that particular value? What I mean is, is 512 somehow special?

Reply:
No, 512 was just off the top of my head.

Reply:
I noticed that you did some benchmarks for this in #6170, so was just wondering how 512 was chosen.

Reply:
Yeah. Maybe this can be changed to a "bring-your-own-`buf: []u8`" function like the rest of the `io.Reader` API, but it may seem weird to pass a buffer to a function which only skips bytes?

Reply:
Yeah, it might seem strange indeed. How about `comptime` config params? Would that make sense? I guess if the dev knows-ish how their streams will be sized, they could then config this with a custom buffer size to use for each `skipBytes` call. Something like:

```zig
try skipBytes(10000, .{ .buf_size = 1024 });
```

Reply:
Oh, and have the default set at 512 or something.